src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,117 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
 import { TextField } from '../TextField';
-
-export const NewMovie = () => {
+import { Movie } from '../../types/Movie';
+// eslint-disable-next-line max-len, prettier/prettier, @typescript-eslint/no-unused-vars
+// const pattern = /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+// const isValid = (url: string) => {
+//   return pattern.test(url);
+// };
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
   const [count] = useState(0);
+  const [title, setTitle] = useState('');
+  const [description, setDescription] = useState('');
+  const [imgUrl, setImgUrl] = useState('');
+  const [imdbUrl, setImdbUrl] = useState('');
+  const [imdbId, setImdbId] = useState('');
```

Comment:
It's not advisable to generate such a large number of states when we only have one instance (movie). 
It'd be much better to utilize a single state with an object inside.

1. Create a variable outside the Component
```
const initialMovieState = {
  title: '',
  description: '',
  imgUrl: '',
  imdbUrl: '',
  imdbId: ''
}
```

2. Use one state for movie fields
3. Pass `initialMovieState` object when you invoke reset `function`
4. You will reuse one handler instead of handlers for each input
Example: 

```
const handleInputChange = (e: ...) => {
  const {name, value} = e.target; setMovie(prevInputs => ({...prevInputs, [name]: value}))
}
```

5. for reset function you will just pass initialMovieState


```suggestion
  const [movie, setMovie] = useState(initialMovieState);
```